client/app/host/AnnoucementForm.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { useToast } from '@/hooks';
+import { Send } from 'lucide-react';
+import { announcementsAtom } from '@/lib/host-state';
+import { useAtom } from 'jotai';
+import { useState, useRef } from 'react';
+import { ipAtom } from '@/lib/services/api';
+import { tokenAtom } from '@/lib/services/auth';
+
+export default function AnnouncementForm() {
+    const containerRef = useRef<HTMLDivElement | null>(null);
+    const [ip] = useAtom(ipAtom);
+    const [authToken] = useAtom(tokenAtom);
+    const [announcementList, setAnnouncements] = useAtom(announcementsAtom);
+    const { toast } = useToast();
+    const [input, setInput] = useState('');
+
+    const postAnnouncement = async (ip: string | null, message: string) => {
+        try {
+            const res = await fetch(`${ip}/announcements`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    Authorization: `Bearer ${authToken}`,
+                },
+                body: JSON.stringify({ message }),
+            });
+            if (!res.ok) {
+                toast({
+                    variant: 'destructive',
+                    title: 'Error',
+                    description: 'An error occured when POSTing to the endpoint.',
+                });
+            }
+            const result = await res.json();
+            setAnnouncements((prev) => [...prev, result]);
+            toast({
+                title: 'Announcement Poted',
+                description: `Message: ${result.message}`,
```

Comment:
The description is not necessary as it's already on another part of the UI.
```suggestion
                title: 'Announcement Sent',
```